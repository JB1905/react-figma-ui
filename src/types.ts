import { colorNames, iconNames, sizes, tints, weights } from './constants';

export type ColorName = (typeof colorNames)[number];
export type IconName = (typeof iconNames)[number];
export type Size = (typeof sizes)[number];
export type Tint = (typeof tints)[number];
export type Weight = (typeof weights)[number];
