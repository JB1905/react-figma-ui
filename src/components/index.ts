export * from './Button';
export * from './Checkbox';
export * from './Disclosure';
export * from './Icon';
export * from './IconButton';
export * from './Input';
export * from './Label';
export * from './Onboarding';
export * from './Radio';
export * from './SectionTitle';
export * from './SelectMenu';
export * from './Switch';
export * from './Textarea';
export * from './Type';
