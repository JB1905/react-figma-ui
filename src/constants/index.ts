export * from './colorNames';
export * from './iconNames';
export * from './sizes';
export * from './tints';
export * from './weights';
