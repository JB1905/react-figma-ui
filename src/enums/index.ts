export { Size } from './Size';
export { Tint } from './Tint';
export { Weight } from './Weight';
