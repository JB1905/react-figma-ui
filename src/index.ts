export { Button } from './components/Button';
export { Checkbox } from './components/Checkbox';
export { Disclosure } from './components/Disclosure';
export { Icon } from './components/Icon';
export { IconButton } from './components/IconButton';
export { Input } from './components/Input';
export { Label } from './components/Label';
export { Onboarding } from './components/Onboarding';
export { Radio } from './components/Radio';
export { SectionTitle } from './components/SectionTitle';
export { SelectMenu } from './components/SelectMenu';
export { Switch } from './components/Switch';
export { Textarea } from './components/Textarea';
export { Type } from './components/Type';
