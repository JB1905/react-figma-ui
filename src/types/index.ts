import { colorNames } from '../constants/colorNames';
import { iconNames } from '../constants/iconNames';
import { sizes } from '../constants/sizes';
import { tints } from '../constants/tints';
import { weights } from '../constants/weights';

export type ColorName = typeof colorNames;
export type IconName = typeof iconNames;
export type Size = typeof sizes;
export type Tint = typeof tints;
export type Weight = typeof weights;
