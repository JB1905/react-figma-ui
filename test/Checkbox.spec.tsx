import React from 'react';
import { render } from '@testing-library/react';

import { Checkbox } from '../src';

describe('Checkbox', () => {
  it('should render unchecked Checkbox', () => {
    const { container } = render(
      <Checkbox id="rick-astley">Never gonna give you up...</Checkbox>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render checked Checkbox', () => {
    const { container } = render(
      <Checkbox id="rick-astley" checked>
        Never gonna give you up...
      </Checkbox>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render disabled Checkbox', () => {
    const { container } = render(
      <Checkbox id="rick-astley" disabled>
        Never gonna give you up...
      </Checkbox>
    );

    expect(container).toMatchSnapshot();
  });
});
