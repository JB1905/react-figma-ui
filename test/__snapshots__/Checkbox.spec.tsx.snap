// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render checked Checkbox 1`] = `
<div>
  <div
    class="checkbox "
  >
    <input
      checked=""
      class="checkbox__box undefined"
      id="rick-astley"
      type="checkbox"
    />
    <label
      class="checkbox__label "
      for="rick-astley"
    >
      Never gonna give you up...
    </label>
  </div>
</div>
`;

exports[`Checkbox should render disabled Checkbox 1`] = `
<div>
  <div
    class="checkbox "
  >
    <input
      class="checkbox__box undefined"
      disabled=""
      id="rick-astley"
      type="checkbox"
    />
    <label
      class="checkbox__label "
      for="rick-astley"
    >
      Never gonna give you up...
    </label>
  </div>
</div>
`;

exports[`Checkbox should render unchecked Checkbox 1`] = `
<div>
  <div
    class="checkbox "
  >
    <input
      class="checkbox__box undefined"
      id="rick-astley"
      type="checkbox"
    />
    <label
      class="checkbox__label "
      for="rick-astley"
    >
      Never gonna give you up...
    </label>
  </div>
</div>
`;
